test/addons-napi/3_callbacks/test.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const common = require('../../common');
+const assert = require('assert');
+const addon = require(`./build/${common.buildType}/binding`);
+
+addon.RunCallback(function(msg) {
+  assert.strictEqual(msg, 'hello world');
+});
+
+const global = function() { return this; }.apply();
+
+function testRecv(desiredRecv) {
+  addon.RunCallbackWithRecv(function() {
+    assert.strictEqual(this,
+                       (desiredRecv === undefined ||
+                        desiredRecv === null) ?
+                       global : desiredRecv);
```

Comment:
It turns out this test was broken in multiple ways, so that it didn't test what it claimed to. I'm fixing it up. In this case, there was never any exception because the native callback encountered and ignored a `napi_invalid_arg`.

Normally, any exception thrown by JS code invoked by a native callback should get automatically propagated back up the JS stack, as long as it is not explicitly cleared by the native code.